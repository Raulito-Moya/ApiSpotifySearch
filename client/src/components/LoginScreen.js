import React from 'react'
import { useDispatch } from 'react-redux'
import { getFetchToken } from '../actions/token';

export const LoginScreen = () => {

const dispatch = useDispatch();
   
  const initState = {
    login: true
  }

    const handleLogin = () => {
      dispatch(getFetchToken())
      localStorage.setItem('login',JSON.stringify(initState))
    }

    return(
        <div>
            <h1>LoginScreen</h1>
            <button 
              type="click"
              onClick={handleLogin}
              >
                Login
            </button>
        </div>
    )
}