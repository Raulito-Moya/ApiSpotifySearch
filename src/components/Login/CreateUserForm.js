import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import Swal from 'sweetalert2';

import { useForm } from '../../Hooks/useForm'
import { createAcount } from '../../actions/logintoken'
import { sendAcountData } from '../../helpers/UserLogin'
import { passwordError } from '../../actions/ui'


export const CreateUserForm = () => {

     const dispatch = useDispatch()
     const { auth } = useSelector( state => state )
  
    const [{nombre,correo, password,confirmPassword}, handleInputChange ] = useForm({
        nombre:'',
        correo: '',
        password: '',
        confirmPassword:''
    })


    const handleCreateUser = async(e) => {
 
      try {
        
        console.log(nombre,correo,password);
        e.preventDefault()
           if(password !== confirmPassword ){
             Swal.fire('Error', 'Las contrasenas deben ser igaules','error')
           }else if(nombre.length <= 2){
            Swal.fire('Error', 'El nombre deben ser un nombe valido','error')
           } else if(correo.length <= 7){
             Swal.fire('Error', 'El correo debe ser un correo valido','error')
           } else{
             dispatch(createAcount({nombre,correo,password})) 
             localStorage.setItem('login', JSON.stringify(auth.login))
           }


      } catch (error) {
        console.log(error);
      }

     
             
         
    }




  return(

    <div className="row item">
         
      <form type='submit' className="col-sm-12">
          <h2>Create a Account</h2>
          <label className="form-label user">Name</label>
          <input type="click" name="nombre" value={nombre} onChange={handleInputChange} className="form-control name" placeholder="nombre"/>

          <label className="form-label user">Correo</label>
          <input type="click" name="correo" value={correo} onChange={handleInputChange}  className="form-control" placeholder="name@example.com"/>
          <label className="form-label user">Password</label>
          <input type="click" name="password" value={password} onChange={handleInputChange} className="form-control " placeholder="password"/>
          <input type="click" name="confirmPassword" value={confirmPassword} onChange={handleInputChange} className="form-control " placeholder="Confirm Password"/>


          <button type="submit" onClick={handleCreateUser} className="buttom">LoginUser</button>
       </form>
       
    </div>


  )



}