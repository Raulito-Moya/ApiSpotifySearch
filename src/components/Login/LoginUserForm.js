import React from 'react'
import { useForm } from '../../Hooks/useForm'

export const LoginFormUser = () => {
   
    const [{correo, password}, handleInputChange ] = useForm({
        correo: '',
        password: ''
    })


    const handleLogin = (e) => {
        e.preventDefault()
        console.log(correo, password);
    }
       
  return(

    <div className="row item">
         
      <form type='submit' className="col-sm-10">
          <h2>Login</h2>
          <label className="form-label user">Correo</label>
          <input className="form-control name" type="click" name="correo" value={correo}  onChange={handleInputChange} placeholder="correo"/>

          <label className="form-label user">Password</label>
          <input className="form-control" type="click" name="password" value={password} onChange={handleInputChange}  placeholder="Password"/>
          
          <button type="submit" onClick={handleLogin} className="buttom">LoginUser</button>
       </form>
       
    </div>


  )



}