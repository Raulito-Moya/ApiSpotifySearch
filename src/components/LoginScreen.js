import axios from 'axios';
import React from 'react'
import { useDispatch } from 'react-redux'
import { getFetchToken } from '../actions/token';

export const LoginScreen = () => {

const dispatch = useDispatch();
   
  const initState = {
    login: true
  }

    const handleLogin = () => {
      dispatch(getFetchToken())
      localStorage.setItem('login',JSON.stringify(initState))
    }

    
  /*  const handleLoginUser = () => {
       const url = "https://accounts.spotify.com/authorize"
       const method = 'get'

      axios.get(url,{
        method,
        params:{
          client_id:'1de72a570fb94a9fa7dbab6d8dd16c24',
          response_type:'code',
          redirect_uri:'https://spotifysearch.com/callback/'
        }
        
      }).then(e => console.log(e))


    }*/

    return(
        <div>
            <h1>LoginScreen</h1>
            <button 
              type="click"
              onClick={handleLogin}
              >
                Login
            </button>

         
        </div>
    )
}