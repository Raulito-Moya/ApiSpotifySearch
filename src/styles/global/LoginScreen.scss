
.loginScreen{
   display: flex;
   flex-direction: column;
}

   
.introForm{
 border-bottom: 0.5px black solid;
 font-size: 50px;
 margin-top:45px;
 //padding-left: 5%;
 padding-bottom: 20px;

}

.item{ 
  margin-top: 50px;
  margin-left:auto;
  margin-right:auto;
  padding-bottom: 190px;
}

.user{
font-size: 26px;
}

.password{
font-size: 26px;
}

.buttom{
 display: block;
 margin-top: 40px;
 /*margin-left:50%;
 margin-right:auto;*/
 background: #ffffff;
 color: rgb(0, 0, 0);
 padding: 10px 15px;
 border-radius: 5px;
 border: none;
 border-bottom: 4px solid #000000;
 border-right: 3px solid #000000;
 border-left: 1px solid #000000;
 border-top: 1px solid #000000;
 
  
}