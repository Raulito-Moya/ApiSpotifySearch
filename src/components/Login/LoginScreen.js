import axios from 'axios';
import React, { useEffect, useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { getFetchToken, getAuthorize } from '../../actions/logintoken';
import { CreateUserForm } from './CreateUserForm';
import { LoginFormUser } from './LoginUserForm';

export const LoginScreen = () => {



  const initState = {
    login: true
  }
  const { ui } = useSelector(state => state)   



    const handleLoginSpotify = () => {
      const data = getAuthorize() 
    
    //  console.log(data);
     // dispatch(getFetchToken())
      //localStorage.setItem('login',JSON.stringify(initState))
    }

    


      const [form, setForm] = useState(JSON.parse(localStorage.getItem('formvalue')) )
      let formValue = 'value' //el value para renderear cada formulario

      const setFormStatus = () => { 
        setForm(!form) 
        console.log(form); 
        }

        JSON.stringify(localStorage.setItem('formvalue',form )) //aqui mando esto fuera del setFormSatus ya que si lo mnado dentro no me coge el valor cambioado por el setState
       formValue = JSON.parse(localStorage.getItem('formvalue')) 
       console.log(formValue);
 


    return(
      <div className="row">
        
        <div className="loginScreen">
           
          { formValue && <LoginFormUser/> }

          { !formValue && <CreateUserForm/> } 
         
          {ui.msg && <h5>Usuario o Contrasena incorrectos</h5>}
         <div className="buttom-change">
           <button type="click" className="btn"  onClick={setFormStatus}>{formValue ? 'Create a Acount ':'Login User'}</button>
         </div>
         
          
          <div className="item">
            <h4>Login with Spotify</h4>
            <button 
             type="click"
             onClick={handleLoginSpotify}
            >
              <img src="/img/spotify-logo.png" alt=""/>
            </button>
          </div>

        </div>
      </div>
    )
}