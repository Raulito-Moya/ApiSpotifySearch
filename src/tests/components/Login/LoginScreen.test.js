import { mount } from "enzyme"
import React from 'react'
import { Provider } from "react-redux"
import thunk from "redux-thunk"
import configureStore from 'redux-mock-store'

import { LoginScreen } from "../../../components/Login/LoginScreen"





const middlewares = [thunk]
const mockstore = configureStore( middlewares )


const initState = { ui:false }
let store = mockstore(initState)
store.dispatch = jest.fn();




const wrapper = mount(
    <Provider store={store}>
        <LoginScreen />
    </Provider>
)


/*jest.mock('../../../components/Login/LoginScreen', () => ({
    setFormStatus: jest.fn(),
   
  
}))*/


describe('Login Screen tests',()=> {

  test('debe renderiar el form correcto',() => {
  
     expect(wrapper).toMatchSnapshot()
     
  
  })
  
  
  test('debe de renderizar el LoginForm',() => {
     
     wrapper.find('button[className="btn"]').simulate('click')
    expect(wrapper).toMatchSnapshot()
  
  
  })
  
  
  test('debe de renderizar el CreateUserform',() => {
     
     wrapper.find('button[className="btn"]').simulate('click')
    expect(wrapper).toMatchSnapshot()
  
  
  })



})