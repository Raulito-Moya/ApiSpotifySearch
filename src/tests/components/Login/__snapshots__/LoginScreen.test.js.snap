// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen tests debe de renderizar el CreateUserform 1`] = `
<div
  className="row"
>
  <div
    className="loginScreen"
  >
    <div
      className="row item"
    >
      <form
        className="col-sm-12"
        type="submit"
      >
        <h2>
          Create a Account
        </h2>
        <label
          className="form-label user"
        >
          Name
        </label>
        <input
          className="form-control name"
          name="nombre"
          onChange={[Function]}
          placeholder="nombre"
          type="click"
          value=""
        />
        <label
          className="form-label user"
        >
          Correo
        </label>
        <input
          className="form-control"
          name="correo"
          onChange={[Function]}
          placeholder="name@example.com"
          type="click"
          value=""
        />
        <label
          className="form-label user"
        >
          Password
        </label>
        <input
          className="form-control "
          name="password"
          onChange={[Function]}
          placeholder="password"
          type="click"
          value=""
        />
        <input
          className="form-control "
          name="confirmPassword"
          onChange={[Function]}
          placeholder="Confirm Password"
          type="click"
          value=""
        />
        <button
          className="buttom"
          onClick={[Function]}
          type="submit"
        >
          LoginUser
        </button>
      </form>
    </div>
    <div
      className="buttom-change"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="click"
      >
        Login User
      </button>
    </div>
    <div
      className="item"
    >
      <h4>
        Login with Spotify
      </h4>
      <button
        onClick={[Function]}
        type="click"
      >
        <img
          alt=""
          src="/img/spotify-logo.png"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Login Screen tests debe de renderizar el LoginForm 1`] = `
<div
  className="row"
>
  <div
    className="loginScreen"
  >
    <div
      className="row item"
    >
      <form
        className="col-sm-10"
        type="submit"
      >
        <h2>
          Login
        </h2>
        <label
          className="form-label user"
        >
          Correo
        </label>
        <input
          className="form-control name"
          name="correo"
          onChange={[Function]}
          placeholder="correo"
          type="click"
          value=""
        />
        <label
          className="form-label user"
        >
          Password
        </label>
        <input
          className="form-control"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="click"
          value=""
        />
        <button
          className="buttom"
          onClick={[Function]}
          type="submit"
        >
          LoginUser
        </button>
      </form>
    </div>
    <div
      className="buttom-change"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="click"
      >
        Create a Acount 
      </button>
    </div>
    <div
      className="item"
    >
      <h4>
        Login with Spotify
      </h4>
      <button
        onClick={[Function]}
        type="click"
      >
        <img
          alt=""
          src="/img/spotify-logo.png"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Login Screen tests debe renderiar el form correcto 1`] = `
<div
  className="row"
>
  <div
    className="loginScreen"
  >
    <div
      className="row item"
    >
      <form
        className="col-sm-12"
        type="submit"
      >
        <h2>
          Create a Account
        </h2>
        <label
          className="form-label user"
        >
          Name
        </label>
        <input
          className="form-control name"
          name="nombre"
          onChange={[Function]}
          placeholder="nombre"
          type="click"
          value=""
        />
        <label
          className="form-label user"
        >
          Correo
        </label>
        <input
          className="form-control"
          name="correo"
          onChange={[Function]}
          placeholder="name@example.com"
          type="click"
          value=""
        />
        <label
          className="form-label user"
        >
          Password
        </label>
        <input
          className="form-control "
          name="password"
          onChange={[Function]}
          placeholder="password"
          type="click"
          value=""
        />
        <input
          className="form-control "
          name="confirmPassword"
          onChange={[Function]}
          placeholder="Confirm Password"
          type="click"
          value=""
        />
        <button
          className="buttom"
          onClick={[Function]}
          type="submit"
        >
          LoginUser
        </button>
      </form>
    </div>
    <div
      className="buttom-change"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="click"
      >
        Login User
      </button>
    </div>
    <div
      className="item"
    >
      <h4>
        Login with Spotify
      </h4>
      <button
        onClick={[Function]}
        type="click"
      >
        <img
          alt=""
          src="/img/spotify-logo.png"
        />
      </button>
    </div>
  </div>
</div>
`;
